provisioner/claim-reconciler/reconiler.go

```diff
@@ -0,0 +1,113 @@
+package object_bucket_claim_reconciler
+
+import (
+	"context"
+	"fmt"
+	"github.com/yard-turkey/lib-bucket-provisioner/pkg/apis/objectbucket.io/v1alpha1"
+	"github.com/yard-turkey/lib-bucket-provisioner/provisioner"
+	"k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+	"time"
+)
+
+type objectBucketClaimReconciler struct {
+	client          client.Client
+	provisionerName string
+	provisioner     provisioner.Provisioner
+	retryInterval   time.Duration
+	retryTimeout    time.Duration
+	retryBackoff    int
+}
+
+var _ reconcile.Reconciler = &objectBucketClaimReconciler{}
+
+type ReconcilerOptions struct {
+	RetryInterval time.Duration
+	RetryTimeout  time.Duration
+	RetryBackoff  int
+}
+
+func NewObjectBucketClaimReconciler(c client.Client, name string, provisioner provisioner.Provisioner, options ReconcilerOptions) *objectBucketClaimReconciler {
+	return &objectBucketClaimReconciler{
+		client:          c,
+		provisionerName: strings.ToLower(name),
+		provisioner:     provisioner,
+		retryInterval:   options.RetryInterval,
```

Comment:
how are defaults handled?

Reply:
currently, the caller sets the defaults before calling this function.  I take your point though - it's probably more appropriate that it be handled here.